### DISCUSSION TITLE
update broke install - need help to resolve

### DISCUSSION THREAD

**Discussion by jpeterse** [inventree/InvenTree#3251] [Q&A] [answered]
I'm running Inventree on bare metail (w. gunicorn/supervisor).
After update, the interface now only show the menu bar at top.
Nothing else show up.
I have two issues to resolve.

### Issue 1
Running the server manually in debug mode, I can see the interface again.
I got 215 notifications, which I found out was because product images now require a .preview. file in addition to the existing .thumbnail. files.
I created the files with ImageMagic, which appeared to satisfy the server, and I was able to make the updates I needed, while running the server in debug mode, without additional error notifications occurring.

Today I started the server in debug mode again. This time got 91 new error notifications.
In admin console, each error just has the message:
"An error has been logged by the server."

I don't see any errors on the console of either the back-end worker, or the front-end server.
Despite the error notifications, things appear to be working.

**How can I identify what is generating the errors and resolved them?**

### Issue 2
As for running the server with gunicorn and supervisor, I do see errors in log file, that static files are not found:
Not Found: /static/bootstrap-table/bootstrap-table.min.css
Not Found: /static/bootstrap-table/bootstrap-table.min.js
Not Found: /static/fullcalendar/locales-all.min.js
Not Found: /static/fullcalendar/main.min.js

When I look at the file system they do appear to exist under src/InvenTree/InvenTree/static, e.g.:
src/InvenTree/InvenTree/static$ find . -iname bootstrap-table.min.css
./bootstrap-table/bootstrap-table.min.css
./bootstrap-table/themes/bootstrap-table/bootstrap-table.min.css

**What do I need to change in the setup to make the interface work again?**

**Comment by SchrodingersGat** [comment, marked as answer]
Looks like you may have missed some steps in the update process. Did you run `invoke update` after pulling down the new code?

The changes you are seeing were merged recently in https://github.com/inventree/InvenTree/pull/3185

> How can I identify what is generating the errors and resolved them?

These can be viewed in the admin interface: https://inventree.readthedocs.io/en/latest/settings/logs/

Please navigate to the admin section to see what the errors were.

> I got 215 notifications, which I found out was because product images now require a .preview. file in addition to the existing .thumbnail. files.
> I created the files with ImageMagic, which appeared to satisfy the server, and I was able to make the updates I needed, while running the server in debug mode, without additional error notifications occurring.

These preview images are automatically created when you run `invoke update`

> As for running the server with gunicorn and supervisor, I do see errors in log file, that static files are not found:

Looks like the required static files have not been copied across. Again, running `invoke update` should manage this for you.

----------------------------

Can you please confirm the steps you followed when updating? It looks like some critical steps have been missed!

**Reply by jpeterse** [reply]
I believe I did run the invoke update
I have a script that executes the update steps, that have worked well up till now, which includes a call to invoke update.
I have in the mean time gotten the gunicorn/supervisor part solved, the interface comes up now.

Still don't know what is causing the errors though.
There is really not much to see in the admin page as far as the error message go.
All it says in the message field is:

> "An error has been logged by the server."

![notification-admin-page](https://user-images.githubusercontent.com/8325689/175753546-67e5b725-9009-4081-afda-518f88a36186.png)
Not much to see on the front end either:
![notification-front-end](https://user-images.githubusercontent.com/8325689/175753549-304fa521-f6a9-4c44-ac81-fcb071afa43a.png)

**Reply by SchrodingersGat** [reply]
You are looking at the wrong page in the admin interface. You are currently looking at "notifications" but you should be looking at "errors"

https://inventree.readthedocs.io/en/latest/settings/logs/

Go to `<server>/admin/error_report/error/` to see the actual content of the error logs.

> I have in the mean time gotten the gunicorn/supervisor part solved, the interface comes up now.

What did you do to solve this part of the problem?

**Reply by jpeterse** [reply]
Ahh, thanks.

So, the 90 errors was for the static file location. It's the same few files there just appear multiple times.
I resolved that earlier today by running invoke update again.
And they those don't appear any more

But 2 new ones do:
/home/inventree/data/static/script/chartjs-adapter-moment.min.js.map
/home/inventree/data/static/script/qr-scanner.umd.min.js.map

This is after running invoke update

**Reply by SchrodingersGat** [reply]
Sounds like it is all working fine now then :) The .map files are not required, you can ignore those warnings
